### PR TITLE
general tweaks

### PR DESCRIPTION
**Pull request by bfgray3**
I haven't worked on any specific issues yet, but I was able to start looking at the code.

### CODE REVIEW

File: R/vi_model.R [hunk 1]
```diff
@@ -49,9 +49,9 @@ vi_model.C5.0 <- function(object, ...) {
   )
 
   # Add variable importance type attribute
-  dots <- list(...)
-  attr(tib, "type") <- if ("metric" %in% names(dots)) {
-    dots[["metric"]]
+  dot_pairlist <- substitute(...())
```
Comment (bfgray3) [inline R/vi_model.R]: We can get away list not evaluating all elements of `...` since we only need `metric` if it's there.

**Comment by bgreenwell** [comment]
Thanks! Where did you pick up the use of `substitute(...())`?

**Comment by bgreenwell** [comment]
Thanks! Where did you pick up the use of `substitue(...())`?

**Comment by bfgray3** [comment]
I learned about it recently.  I think it's pretty slick.

``` r
foo <- function(...) substitute(...())
system.time(bar <- foo(x = rnorm(1e100), y = 1))
   user  system elapsed 
      0       0       0 
str(bar)
Dotted pair list of 2
 $ x: language rnorm(1e+100)
 $ y: num 1
str(bar[["y"]])
 num 1
```

Created on 2018-08-06 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

I learned about it on the r-devel mailing list.  Here are some notes I had taken/copy-pasted from the thread.  I think the discussion had started as a way to figure out how to get the length of `...`.

```
New low level utilities ...elt(n) and ...length() for working with ... parts inside a function.

dotlength <- function(...) nargs()

(This is subtly different from calling nargs() directly as it will
only count the elements in ...)

dddlen <- function(...)length(match.call(expand.dots=FALSE)$...)

You can get the first element of ... without evaluating the rest using 
..1, but I don't know a way to do this for general n in pre-3.5.0 base R.

BTW, substitue(...()) is beautiful.

In R-3.5.0 you can use ...length():
  > f <- function(..., n) ...length()
  > f(stop("one"), stop("two"), stop("three"), n=7)
  [1] 3

Prior to that substitute() is the way to go
  > g <- function(..., n) length(substitute(...()))
  > g(stop("one"), stop("two"), stop("three"), n=7)
  [1] 3

R-3.5.0 also has the ...elt(n) function, which returns
the evaluated n'th entry in ... , without evaluating the
other ... entries.
  > fn <- function(..., n) ...elt(n)
  > fn(stop("one"), 3*5, stop("three"), n=2)
  [1] 15

Prior to 3.5.0, eval the appropriate component of the output
of substitute() in the appropriate environment:
  > gn <- function(..., n) {
  +   nthExpr <- substitute(...())[[n]]
  +   eval(nthExpr, envir=parent.frame())
  + }
  > gn(stop("one"), environment(), stop("two"), n=2)
  <environment: R_GlobalEnv>
```

**Comment by bfgray3** [comment]
Actually, I need to `eval` `dot_pairlist[["metric"]]`.  I'll make a PR later today.  Sorry for the mistake.

**Comment by bgreenwell** [comment]
No worries, thanks for the info!

**Pull request merged**